uniffi/src/interface/mod.rs

```diff
@@ -270,14 +293,15 @@ impl FromStr for ComponentInterface {
     }
 }
 
+/// Trait to help build a `ComponentInterface` from WedIDL syntax nodes.
+///
+/// This trait does structural matching on the various weedle AST nodes and
+/// uses them to build up the records, enums, objects etc in the provided
+/// `ComponentInterface`.
 trait APIBuilder {
     fn process(&self, ci: &mut ComponentInterface) -> Result<()>;
 }
 
-trait APIConverter<T> {
-    fn convert(&self, ci: &ComponentInterface) -> Result<T>;
-}
-
```

Comment:
I moved some trait implementations around to put them closer to the structs they operate on, in the hope of making it easier to digest the code one piece at a time.